config/charts/prometheus-operator.yaml

```diff
@@ -47,6 +47,11 @@ alertmanager:
     - name: 'webhook'
       webhook_configs:
       - url:  "http://matrixbot:8080/skill/alertmanager/webhook"
+    - name: opsGenie-receiver
+      opsgenie_configs:
+      - api_key: {{ env "OPSGENIE_TOKEN" }}
```

Comment:
There's no helmfile here, so `env` won't work. I think we can define our own values and connect them with configuration settings, for instance, in the config:
```
{
...
  "opsgenie": true
....
}
```
then in config/charts/prometheus-operator.yaml:
```
{{ if .Values.opsgenie.enabled }}
- name: opsGenie-receiver
      opsgenie_configs:
      - api_key: {{ .Values.opsgenie.token }}
        api_url: {{ .Values.opsgenie.url }}
{{ end }}
```
and in lib/clients/helm.js, replace `_installPrometheusOperator` content with something like `_installMatrixbot`, and set the values depending on the configuration settings and env vars:
```
async _installPrometheusOperator() {                                                                                                                                                                                      
    await this._cmd(`upgrade --install prometheus-operator --set opsgenie.enabled=${this.config.opsgenie} --set opsgenie.token=${this.opsgenieToken} --set opsgenie.url=${this.opsgenieUrl} --version ${this.mainConfig.chartVersions['prometheus-operator']} stable/prometheus-operator`);                                                                                                                                                                  
  }   
```
WDYT?

Reply:
I thought that it is better to add the opsgenie through the environmental variables like so.
WDYT?

doubt that it will work as-is. trying to find out how to debug nodejs.

Reply:
https://github.com/w3f/polkadot-deployer/blob/Setup-opsgenie-integration-with-Kusama-core-infrastructure/lib/clients/helm.js#L392-L399 looks good, but we should only leave credentials to env vars, we configure the creation through the config file, this is where we define declaratively how the deployment should look like. So I would remove `OPSSGENIE_ENABLED` env var and i would use instead a config param, let me know  wdyt.

Reply:
I moved the test to line https://github.com/w3f/polkadot-deployer/blob/83fb5145dc5b2aa4811823418b6cd27e0fa1c818/lib/clients/helm.js#L391

if token is set, the opsgenie will be enabled.
updated readme also. Now If I could just get it to run...

Reply:
No, this way we are still using an env var to control if we add opsgenie settings, this should be done with a configuration option. We need to have the description of the network as something that can be added to version control, that can be reviewed, rolled back, etc.